bundles/net.tourbook/src/net/tourbook/database/TourDatabase.java

```diff
@@ -3111,12 +3115,20 @@ private void createTable_TourData(final Statement stmt) throws SQLException {
             //
             // version 40 end
 
-            // version 41 start  -  20.5
+            // version 41 start  -  20.8
             //
             + " maxPace                               FLOAT DEFAULT 0,                  \n" //$NON-NLS-1$
             //
             // version 41 end
 
+            // version 42 start  -  20.X
+            //
+            + " tourTimerPauses                               BLOB,                  \n" //$NON-NLS-1$
```

Comment:
As documented in line 3130...3137 there was a fatal issue when using more than one BLOB field

Reply:
`tourTimerPauses` are also defined as EJB entity

Reply:
I'm wondering why an EJB is used for the paused times which also could be a list in `net.tourbook.data.SerieData` as it is currenly only a pause start/end time list, similar to other data series.

An additional sql join will propably reduce performance.

Reply:
I thought about using `SerieData `but was not sure what was the best. 
Do you prefer that I use `SerieData `? I can modify it.

Reply:
When there are no additional data for a paused time "slice" (e.g. a tour marker has many additional fields) then we should go with `SerieData`

Reply:
Ok, I tried to have the tour timer pauses as a serie data property like this :
public TourTimerPause[] timerPausesSerie;

But the database saving fails as below :

Is it maybe because it is not an array of a primitive type like an `int` or `float` ?

```
Caused by: java.io.NotSerializableException: net.tourbook.data.TourData
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.writeArray(ObjectOutputStream.java:1378)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1174)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
	at org.hibernate.util.SerializationHelper.serialize(SerializationHelper.java:125)
	... 21 more
```

Reply:
I think when you serialize `TourTimerPause` then it's contained `TourData` would also be serialized which is not a good idea.

So the idea behind `SerieData` is that it contains mass data series and not serialized Objects, serialized Object should be saved as EJB.

In this case I would put 2 new int[] into `SerieData`, e.g.
```
int[] pausedTime_Start
int[] pausedTime_End
```

These new field's may not contain many values compared with the other data serie values but to reduce diskspace, only the necessary data type should be used, so instead of `long` as in `TourTimerPause`, `int` would be sufficient.

I know that `latitude` and `longitude` in `SerieData` contain a lot of space but since some years I have the idea to reduce them sometimes to a much shorter footprint, like https://stackoverflow.com/questions/5664930/from-latitude-and-longitude-to-latitudee6-longitudee6 but this is a lot of work :-)

Reply:
Good idea, I have implemented that.

I can't find much info about what `longitudee6 `is. Is it just an `int `instead of a `double `?

Reply:
E6 is an `int` with latitude/longitude * 10^6, I think that I saw it first in the 2.5D map code

There are a few remainings in the MT code

![mt-e6](https://user-images.githubusercontent.com/1283445/92555811-cc84d780-f268-11ea-9085-b139659d7610.png)

Reply:
2.5D map code https://appdoc.app/artifact/org.mapsforge/vtm/0.9.0/org/oscim/core/GeoPoint.html#latitudeE6